src/pages/ProfilePage.js

```diff
@@ -1,16 +1,128 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { Redirect } from "react-router-dom";
 
 import Container from "@material-ui/core/Container";
+import Divider from "@material-ui/core/Divider";
+import Grid from "@material-ui/core/Grid";
+import { makeStyles } from "@material-ui/core/styles";
 import Paper from "@material-ui/core/Paper";
 import Typography from "@material-ui/core/Typography";
+import EqualizerIcon from "@material-ui/icons/Equalizer";
 
-import User from "../components/User";
+import ProfileName from "../components/ProfileName";
+import UserStatistics from "../components/UserStatistics";
+import ProfileGamesTable from "../components/ProfileGamesTable";
+import useFirebaseRef from "../hooks/useFirebaseRef";
+import firebase from "../firebase";
+import { computeState } from "../util";
+import LoadingPage from "./LoadingPage";
+
+const useStyles = makeStyles((theme) => ({
+  divider: {
+    [theme.breakpoints.down("sm")]: {
+      display: "none",
+    },
+  },
+}));
 
 function ProfilePage({ match }) {
+  const userId = match.params.id;
+
+  const [games, loadingGames] = useFirebaseRef(`/userGames/${userId}`);
+  const [gamesData, setGamesData] = useState(null);
+  const [redirect, setRedirect] = useState(null);
+
+  const classes = useStyles();
+
+  const handleClickGame = (gameId) => {
+    setRedirect(`/room/${gameId}`);
+  };
+
+  useEffect(() => {
+    function mergeGameData(game, gameData) {
+      const scores = computeState(gameData).scores;
+      const winner = Object.keys(game.users).sort((u1, u2) => {
+        const s1 = scores[u1] || 0;
+        const s2 = scores[u2] || 0;
+        if (s1 !== s2) return s2 - s1;
+        return u1 < u2 ? -1 : 1;
+      })[0];
+      return {
+        ...game,
+        ...gameData,
+        winner: winner,
+        scores: scores,
+      };
+    }
+    async function getGamesData() {
+      const gameReads = [];
+      const gameDataReads = [];
+      for (const gameId of Object.keys(games || {})) {
+        gameReads.push(
+          firebase.database().ref(`games/${gameId}`).once("value")
```

Comment:
I don't like what's going on with this asynchronous `once()` code here, especially since you're calling it on every `useEffect()`. I think this could lead to data races between multiple effects if the `games` object changes at just the right time.

Will think about how to fix this tomorrow.